### PR TITLE
Implicit definitions for kore.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**

### CODE REVIEW

**Comment by virgil-serbanuta** [comment]
Hi, Xiaohong, I added you in case you wanted to take a look at kore.kore

**Comment by hreada** [comment]
`ImplicitKore` looks good in general. Be prepared to change the axioms of the proof system, though, as we now have a really nice & minimal proof system, and we will use this new minimal proof system in the future.

**Comment by virgil-serbanuta** [comment]
Jenkins: test this please.

File: src/main/haskell/language-kore/src/Data/Kore/Parser/LexemeImpl.hs [hunk 1]
```diff
@@ -123,21 +123,34 @@ koreKeywordsSet :: Trie.Trie ()
...
+genericIdRawParser firstCharSet bodyCharSet idKeywordParsing = do
     c <- Parser.peekChar'
     idChar <- if not (c `CharSet.elem` firstCharSet)
         then fail ("genericIdRawParser: Invalid first character '" ++ c : "'.")
         else Parser.takeWhile (`CharSet.elem` bodyCharSet)
     let identifier = Char8.unpack idChar
-    when (isJust $ Trie.lookup idChar koreKeywordsSet)
-        (fail ("Identifiers should not be keywords: '" ++ identifier ++ "'."))
+    case idKeywordParsing of
+        KeywordsForbidden ->
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/LexemeImpl.hs]: consider folding the case above in the when instruction below

**Pull request merged**